src/librepgp/stream-parse.c

```diff
@@ -1063,8 +1076,11 @@ init_encrypted_src(pgp_processing_ctx_t *ctx, pgp_source_t *src, pgp_source_t *r
                 goto finish;
             }
         } while (1);
-    } else {
-        errcode = RNP_ERROR_NOT_SUPPORTED;
+    }
+
+    if (!have_key) {
+        (void) fprintf(stderr, "%s: failed to obtain decrypting key or password\n", __func__);
```

Comment:
We should try to return an appropriate error code here. Currently this logs to the console when I use an incorrect passphrase, for example. I believe this function would be used by FFI clients, so ideally we would not do that here, but at the call site.